subsys/bluetooth/controller/rng_driver.c

```diff
@@ -31,11 +33,23 @@ static int rng_driver_get_entropy(struct device *dev, u8_t *buf, u16_t len)
 	u8_t *p_dst = buf;
 	u16_t bytes_left = len;
 	while (bytes_left > 0) {
-		int32_t bytes_read;
-		while ((bytes_read = ble_controller_rand_vector_get(p_dst, bytes_left)) <= 0) {
-			/* Put the thread on wait until next interrupt to get more
-			 * random values. */
-			k_sem_take(&rng_data.sem_sync, K_FOREVER);
+		int32_t bytes_read = 0;
+		while (bytes_read <= 0) {
+			int32_t errcode = MULTITHREADING_LOCK_ACQUIRE_FOREVER_WAIT();
```

Comment:
Why is FOREVER_WAIT needed here?

Reply:
Decided to use regular ACQUIRE macro for rng_driver_get_entropy() and return immediately on error to prevent lock out.